### PR TITLE
blackmagic-desktop-video,decklink: 12.5a15 -> 12.7.1a1

### PR DESCRIPTION
**Pull request by hexchen**
## Description of changes

Update of the decklink driver to the latest version. Driver compiles on the latest kernel as found in `linuxPackages_latest`. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/decklink/default.nix [hunk 1]
```diff
@@ -18,7 +18,7 @@ stdenv.mkDerivation rec {
   nativeBuildInputs =  kernel.moduleBuildDependencies;
 
   postUnpack = ''
-    tar xf Blackmagic_Desktop_Video_Linux_${lib.versions.majorMinor version}/other/${stdenv.hostPlatform.uname.processor}/desktopvideo-${version}-${stdenv.hostPlatform.uname.processor}.tar.gz
+    tar xf Blackmagic_Desktop_Video_Linux_${lib.versions.majorMinor version}.${lib.versions.patch version}/other/${stdenv.hostPlatform.uname.processor}/desktopvideo-${version}-${stdenv.hostPlatform.uname.processor}.tar.gz
```
Comment (hexchen) [inline pkgs/os-specific/linux/decklink/default.nix:21]: I fear this will break again if we get a release without a patch...

**Comment by reinismu** [comment]
Really looking forward to this :) 

Might not exactly be related to this update, but have you encountered driver crashing related to audio? 

<details>
  <summary>Dmesg output</summary>
  
```
[    1.826633] blackmagic: loading out-of-tree module taints kernel.
[    1.826638] blackmagic: module license 'Proprietary' taints kernel.
[    1.826639] Disabling lock debugging due to kernel taint
[    1.827587] blackmagic: Loading driver (version: 12.5a15)
[    1.872659] BlackmagicIO: Driver version 12.5a15 loading
[    1.872694] blackmagic-io 0000:06:00.0: enabling device (0000 -> 0002)
[    1.873058] BlackmagicIO: WARNING: "DeckLink Quad HDMI Recorder" has a firmware version mismatch (valid: 1 device: 0x011b driver: 0x0121)
[    1.873068] BlackmagicIO: Enabled device "DeckLink Quad HDMI Recorder (1)" x4/5 GT/s (471d4ab0,590600) FW Date: 12-31 10:59
[    1.875894] BlackmagicIO: DeckLink Quad HDMI Recorder (1) as blackmagic!io0 [0000:06:00.0]
[    1.875895] BlackmagicIO: Initialised serial driver for blackmagic!io0
[    1.875978] BlackmagicIO: WARNING: "DeckLink Quad HDMI Recorder" has a firmware version mismatch (valid: 1 device: 0x011b driver: 0x0121)
[    1.875988] BlackmagicIO: Enabled device "DeckLink Quad HDMI Recorder (2)" x4/5 GT/s (471d4ab1,590601) FW Date: 12-31 10:59
[    1.878536] BlackmagicIO: DeckLink Quad HDMI Recorder (2) as blackmagic!io1 [0000:06:00.0]
[    1.878537] BlackmagicIO: Initialised serial driver for blackmagic!io1
[    1.878604] BlackmagicIO: WARNING: "DeckLink Quad HDMI Recorder" has a firmware version mismatch (valid: 1 device: 0x011b driver: 0x0121)
[    1.878614] BlackmagicIO: Enabled device "DeckLink Quad HDMI Recorder (3)" x4/5 GT/s (471d4ab2,590602) FW Date: 12-31 10:59
[    1.881143] BlackmagicIO: DeckLink Quad HDMI Recorder (3) as blackmagic!io2 [0000:06:00.0]
[    1.881144] BlackmagicIO: Initialised serial driver for blackmagic!io2
[    1.881192] BlackmagicIO: WARNING: "DeckLink Quad HDMI Recorder" has a firmware version mismatch (valid: 1 device: 0x011b driver: 0x0121)
[    1.881202] BlackmagicIO: Enabled device "DeckLink Quad HDMI Recorder (4)" x4/5 GT/s (471d4ab3,590603) FW Date: 12-31 10:59
[    1.883752] BlackmagicIO: DeckLink Quad HDMI Recorder (4) as blackmagic!io3 [0000:06:00.0]
[    1.883754] BlackmagicIO: Initialised serial driver for blackmagic!io3
[    1.901276] BlackmagicIO: Initialised ALSA driver for blackmagic!io3
[    1.901364] BlackmagicIO: Initialised ALSA driver for blackmagic!io2
[    1.901434] BlackmagicIO: Initialised ALSA driver for blackmagic!io1
[    1.901497] BlackmagicIO: Initialised ALSA driver for blackmagic!io0

[    6.077204] BUG: kernel NULL pointer dereference, address: 0000000000000068
[    6.077207] #PF: supervisor read access in kernel mode
[    6.077209] #PF: error_code(0x0000) - not-present page
[    6.077209] PGD 11fa5f067 P4D 11fa5f067 PUD 11fa5e067 PMD 0
[    6.077212] Oops: 0000 [#1] PREEMPT SMP NOPTI
[    6.077214] CPU: 5 PID: 1625 Comm: wireplumber Tainted: P           O       6.1.61 #1-NixOS
[    6.077216] Hardware name: ASUS System Product Name/TUF GAMING B660M-PLUS D4, BIOS 2014 10/13/2022
[    6.077217] RIP: 0010:_ZN20AudioOutputInterface16startAudioStreamEP27BlackmagicAudioOutputStreamP32BlackmagicAudioOutputStartParams+0x32/0x310 [blackmagic_io]
[    6.077250] Code: e5 41 57 41 56 41 55 41 54 53 48 83 ec 48 44 8b 2a 41 8d 4d fe 83 f9 3e 77 19 49 b9 41 40 00 40 00 00 00 40 49 0f a3 c9 73 09 <0f> b6 4f 68 41 39 cd 76 15 48 83 c4 48 5b 41 5c 41 5d 41 5e 41 5f
[    6.077251] RSP: 0018:ffffa2fec1f6fdb8 EFLAGS: 00010293
[    6.077253] RAX: 00000000000003ea RBX: ffff94854cda1500 RCX: 0000000000000000
[    6.077254] RDX: ffffa2fec1f6fe40 RSI: ffff94854cda1568 RDI: 0000000000000000
[    6.077254] RBP: ffffa2fec1f6fe28 R08: ffff948554a678c8 R09: 4000000040004041
[    6.077255] R10: ffffa2fec1f6fee8 R11: 0000000000000000 R12: 0000000000000002
[    6.077256] R13: 0000000000000002 R14: ffff94855dfade01 R15: 0000000000000000
[    6.077257] FS:  00007f8c3653d040(0000) GS:ffff948c9fb40000(0000) knlGS:0000000000000000
[    6.077258] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    6.077259] CR2: 0000000000000068 CR3: 000000015366e000 CR4: 0000000000750ee0
[    6.077260] PKRU: 55555554
[    6.077260] Call Trace:
[    6.077261]  <TASK>
[    6.077264]  ? __die_body.cold+0x1a/0x1f
[    6.077267]  ? page_fault_oops+0xd2/0x2b0
[    6.077269]  ? exc_page_fault+0x66/0x150
[    6.077271]  ? asm_exc_page_fault+0x22/0x30
[    6.077274]  ? _ZN20AudioOutputInterface16startAudioStreamEP27BlackmagicAudioOutputStreamP32BlackmagicAudioOutputStartParams+0x32/0x310 [blackmagic_io]
[    6.077299]  ? bmio_audio_enable+0x64/0x120 [blackmagic_io]
[    6.077316]  ? ioctl_has_perm.constprop.0.isra.0+0xdd/0x140
[    6.077319]  ? bmio_audio_prepare+0x7a/0x90 [snd_blackmagic_io]
[    6.077322]  ? snd_pcm_do_prepare+0x26/0x40 [snd_pcm]
[    6.077328]  ? snd_pcm_action_single+0x33/0x80 [snd_pcm]
[    6.077333]  ? snd_pcm_action_nonatomic+0x95/0xa0 [snd_pcm]
[    6.077338]  ? snd_pcm_ioctl+0x23/0x40 [snd_pcm]
[    6.077342]  ? __x64_sys_ioctl+0x8d/0xd0
[    6.077344]  ? do_syscall_64+0x37/0x90
[    6.077346]  ? entry_SYSCALL_64_after_hwframe+0x64/0xce
[    6.077348]  </TASK>
[    6.077348] Modules linked in: snd_seq_dummy snd_hrtimer snd_seq snd_seq_device exfat nf_tables libcrc32c nfnetlink msr nouveau snd_sof_pci_intel_tgl snd_sof_intel_hda_common snd_soc_hdac_hda soundwire_intel soundwire_generic_allocation soundwire_cadence snd_sof_intel_hda snd_sof_pci snd_sof_xtensa_dsp snd_sof intel_rapl_msr intel_rapl_common intel_tcc_cooling x86_pkg_temp_thermal intel_powerclamp snd_sof_utils coretemp snd_hda_ext_core crc32_pclmul polyval_clmulni snd_soc_acpi_intel_match polyval_generic gf128mul ghash_clmulni_intel snd_soc_acpi sha512_ssse3 soundwire_bus mxm_wmi sha512_generic drm_ttm_helper aesni_intel ttm cmdlinepart libaes spi_nor crypto_simd eeepc_wmi snd_soc_core cryptd drm_display_helper snd_hda_codec_realtek asus_wmi mtd rapl iTCO_wdt battery ee1004 sparse_keymap intel_pmc_bxt snd_compress snd_hda_codec_generic platform_profile watchdog pmt_telemetry ac97_bus rfkill pmt_class wmi_bmof ledtrig_audio intel_cstate snd_hda_codec_hdmi drm_kms_helper
[    6.077374]  snd_pcm_dmaengine intel_uncore nls_iso8859_1 r8169 snd_hda_intel nls_cp437 agpgart snd_intel_dspcfg vfat i2c_algo_bit realtek fb_sys_fops snd_intel_sdw_acpi uas mdio_devres spi_intel_pci snd_hda_codec mei_me fat syscopyarea i2c_i801 intel_lpss_pci tpm_crb libphy usb_storage spi_intel i2c_smbus snd_hda_core mei video intel_lpss sysfillrect intel_vsec sysimgblt fan thermal idma64 input_leds evdev snd_hwdep joydev tpm_tis virt_dma led_class mousedev mac_hid acpi_tad tpm_tis_core wmi intel_pmc_core pinctrl_alderlake acpi_pad tiny_power_button button sch_fq_codel snd_blackmagic_io(PO) snd_pcm snd_timer snd soundcore blackmagic_io(PO) blackmagic(PO) ctr atkbd libps2 serio vivaldi_fmap loop cpufreq_powersave tun tap macvlan bridge stp llc kvm_intel kvm drm irqbypass fuse deflate backlight configfs efi_pstore efivarfs tpm rng_core dmi_sysfs ip_tables x_tables autofs4 ext4 hid_generic usbhid hid crc32c_generic crc16 mbcache jbd2 sd_mod ahci libahci xhci_pci xhci_pci_renesas libata
[    6.077408]  firmware_class xhci_hcd nvme usbcore nvme_core scsi_mod t10_pi crc32c_intel rtc_cmos crc64_rocksoft crc64 crc_t10dif crct10dif_generic vmd crct10dif_pclmul usb_common scsi_common crct10dif_common dm_mod dax
[    6.077416] CR2: 0000000000000068
[    6.077418] ---[ end trace 0000000000000000 ]---
[    6.084221] r8169 0000:05:00.0 enp5s0: Link is Up - 1Gbps/Full - flow control rx/tx
[    6.084231] IPv6: ADDRCONF(NETDEV_CHANGE): enp5s0: link becomes ready
[    6.179110] RIP: 0010:_ZN20AudioOutputInterface16startAudioStreamEP27BlackmagicAudioOutputStreamP32BlackmagicAudioOutputStartParams+0x32/0x310 [blackmagic_io]
[    6.179156] Code: e5 41 57 41 56 41 55 41 54 53 48 83 ec 48 44 8b 2a 41 8d 4d fe 83 f9 3e 77 19 49 b9 41 40 00 40 00 00 00 40 49 0f a3 c9 73 09 <0f> b6 4f 68 41 39 cd 76 15 48 83 c4 48 5b 41 5c 41 5d 41 5e 41 5f
[    6.179158] RSP: 0018:ffffa2fec1f6fdb8 EFLAGS: 00010293
[    6.179160] RAX: 00000000000003ea RBX: ffff94854cda1500 RCX: 0000000000000000
[    6.179161] RDX: ffffa2fec1f6fe40 RSI: ffff94854cda1568 RDI: 0000000000000000
[    6.179162] RBP: ffffa2fec1f6fe28 R08: ffff948554a678c8 R09: 4000000040004041
[    6.179163] R10: ffffa2fec1f6fee8 R11: 0000000000000000 R12: 0000000000000002
[    6.179164] R13: 0000000000000002 R14: ffff94855dfade01 R15: 0000000000000000
[    6.179164] FS:  00007f8c3653d040(0000) GS:ffff948c9fb40000(0000) knlGS:0000000000000000
[    6.179166] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    6.179166] CR2: 0000000000000068 CR3: 000000015366e000 CR4: 0000000000750ee0
[    6.179167] PKRU: 55555554
...
  
</details>

https://forum.blackmagicdesign.com/viewtopic.php?f=12&t=192214

**Comment by hexchen** [comment]
I had previously encountered that issue, yes. Don't remember how i fixed it though.

note to self: package the supporting utilities, especially the firmware update tool.

**Comment by reinismu** [comment]
Maybe you could bring over all tooling as well? Here is how I did it https://github.com/reinismu/nixpkgs/pull/1

Tho firmware update still didn't fix the driver crash. Talking with Blackmagic support. I hope they will help me solve it

**Comment by hexchen** [comment]
I won't have time to debug the crash in the next 2 weeks or so; I can try to reproduce it after that.

With regards to bringing over the other utilities, including those in the mediaexpress package: i have plans to do so, just haven't had the time yet.

**Comment by reinismu** [comment]
I made a minimal example https://github.com/reinismu/nixos-decklink-ffmpeg from what I got. Sent it to Blackmagic support as well.

Thanks for helping me with this

**Comment by reinismu** [comment]
> I won't have time to debug the crash in the next 2 weeks or so; I can try to reproduce it after that.
> 
> With regards to bringing over the other utilities, including those in the mediaexpress package: i have plans to do so, just haven't had the time yet.

Sorry to poke, but maybe you have recalled how you managed to fix driver crash?

**Comment by hexchen** [comment]
@reinismu i pushed an update in #304621, feel free to test. let me know if your crash still occurs, i currently have my encoder hooked up semi-permanently.

**Pull request closed without merging**